src/tiled/resizehelper.cpp

```diff
@@ -179,7 +177,13 @@ void ResizeHelper::recalculateScale()
     // Pick the smallest scale
     const double scaleW = _size.width() / (double) width;
     const double scaleH = _size.height() / (double) height;
-    mScale = (scaleW < scaleH) ? scaleW : scaleH;
+    mScale = qMin(scaleW, scaleH);
```

Comment:
That the scale is recalculated doesn't mean it'll change, so please still compare with the previous scale and don't recreate the image when it isn't necessary.